.travis.yml

```diff
@@ -14,82 +14,49 @@ git:
 # Global environment variables
 env:
     global:
-        - ARTIFACTS_DIR=build/travis-artifacts
         - DIST=xenial
         - COV=1
         # Default beaver image names. May be overriden in specific stages
         - BEAVER_DOCKER_IMG=builder
         - BEAVER_DOCKER_CONTEXT=docker/builder
         # Make sure beaver is in the PATH
         - PATH="$(git config -f .gitmodules submodule.beaver.path)/bin:$PATH"
-        # We make use of artifact functionality to pass packages between stages.
-        # To avoid packages from old builds accumulated in the artifact dir
-        # additional dummy env key is used so that each build gets own cache.
-        - ID=$TRAVIS_BUILD_ID
 
 # Do a shallow submodule fetch
 before_install: git submodule update --init
 
-# (Ab)use Travis cache to pass artifacts between stages.
-# As long as the environment variables of different stages match, travis will
-# pull the same "cache", so we can use this to actually pass artifacts between
-# stages.
-cache:
-    directories:
-        - $ARTIFACTS_DIR
-
 # Basic config is inherited from the global scope
 jobs:
     templates:
         - &test-matrix
           stage: Test
           after_success: beaver dlang codecov
           script:
+              - set -e # Do not continue on failure
               - beaver dlang install
               - beaver dlang make
-              # Build packages too, this should be done by beaver:
-              # https://github.com/sociomantic-tsunami/beaver/issues/34
+        - &package-matrix
+          stage: Build and upload packages
+          if: tag IS present
+          script:
+              - set -e # Do not continue on failure
+              - beaver dlang install
               - beaver dlang make pkg
-              # Save the packages in the cache directory so
-              # it's accessible in next stages
-              - cp -v build/last/pkg/*.deb "$ARTIFACTS_DIR/";
+              - beaver bintray upload -d sociomantic-tsunami/nodes/dhtnode build/production/pkg/*.deb
```

Comment:
Why `production` and not `last`? Also, maybe we should just go back to use the `deploy:` script in travis instead of a different stage, I'm not sure is worth building twice just for a nicer visualization.

Reply:
> Why production and not last?

Fool-proof in case devel build ends up existing too. We don't want to upload non-production packages!

> Also, maybe we should just go back to use the deploy:

Is it possible though? We don't package/upload every build we test, just one for D1 and D2 each, no matter how big is test matrix

Reply:
With an `if` (or was it `when`?) condition it should.